core/src/server.cpp

```diff
@@ -289,6 +283,27 @@ void Server::allowMessage()
     can_send_ic_messages = true;
 }
 
+void Server::handleDiscordIntegration()
+{
+    if (discord != nullptr) {
+        discord->deleteLater();
```

Comment:
If you're not going to wait for the deletion, just use delete.

Reply:
???. I mark it for deletion and invalidate it's pointer in case someone runs /reload twice in a row. Nothing wrong with that. I could listen for the destroyed signal, but that's not really useful.

Reply:
Who said anything about listening for a destroyed signal? deleteLater() just posts a deletion event that will be handled later. That basically just calls delete. You're not really waiting for discord to be deleted before you go ahead and remake it, and you're not doing this from a slot Discord can call. It's fine to just delete the object outright here. It's probably trivial, but there's no real benefit to using deleteLater() here as far as I can tell. If that is incorrect in this circumstance, then feel free to ignore me. Though in fairness, I don't think the deletion is necessary at all.